Assets/Scripts/EnemyController.cs

```diff
@@ -10,25 +10,27 @@ public class EnemyController : MonoBehaviour
     private int currentHealth;
     [SerializeField]
     private float fireRate = 1f;
+    private float nextFire;
+    private bool inHash = false;
+
     [SerializeField]
     private GameObject projectile;
     [SerializeField]
-    private Transform player;
+    private PlayerController player;
     private Rigidbody2D rb2D;
     private Vector2 targetPosition;
-
-    float nextFire;
+    private Transform playerTransform,
+                      thisTransform;
 
     // Start is called before the first frame update
-    void Start()
-    {
-        
-        
-    }
+    void Start() {  }
 
-    private void Awake() {
+    private void Awake()
+    {
         currentHealth = maxHealth;
         nextFire = Time.time;
+        playerTransform = player.GetComponent<Transform>();
+        thisTransform = this.GetComponent<Transform>();
```

Comment:
`transform` can be used instead of `GetComponent<Transform>()` ([documentation](https://docs.unity3d.com/ScriptReference/Component-transform.html)).